TestResultSummaryService/parsers/BenchmarkParser.js

```diff
@@ -2,169 +2,135 @@ const Parser = require( './Parser' );
-        if (!(Array.isArray(splitOutput)) || splitOutputLength === 1) {
-            return null;
+    // True iff all benchmarks and their corresponding iterations in the given run can be parsed
+    static canParse( testName ) {
+        const {benchmarkName, benchmarkVariant} = this.getBenchmarkInfo(testName);
+        const variantValue = parseInt(benchmarkVariant);
+        if (benchmarkName && benchmarkVariant && BenchmarkMetricRouter[benchmarkName] && BenchmarkMetricRouter[benchmarkName][benchmarkVariant]) {// this works for PerfNext
+            return true;
+        } else if (!isNaN (variantValue) && benchmarkName && benchmarkVariant && BenchmarkMetricRouter[benchmarkName] && BenchmarkMetricRouter[benchmarkName]["0"]) {// this works for TKG
+            return true;
         }
-        return {
-            next: function() {
-                return index < splitOutputLength ?
-                    {value: splitOutput[index++], done: false} :
-                    {done: true};
-            }
-        };
+        return false;
     }
 
-    parse( output ) {
-
-        const benchmarkIterator = BenchmarkParser.getBenchmarkIterator(output);
-        let curItr = benchmarkIterator.next();
-        let testIndex = 1;
+    static perfBuildResult(tests){
         let buildResult;
-        const tests = [];
-
-        while (curItr.done !== true) {
-
-            let curBenchmarkName = null;
-            let curBenchmarkVariant = null;
-            let curBenchVariant = null;
-            let curMetric = null;
-            let curSearchString = null;
-            let curRegexResult = null;
-            let curMetricValues = null;
-            let curProductResource = null;
-            let curRegex = null;
-            let isValid = true;
-            let curTestData = {};
-
-            // Parse benchmark name
-            if ( ( curRegexResult = benchmarkNameRegex.exec( curItr.value ) ) !== null ) {
-                curBenchmarkName = curRegexResult[1];
+        if ((tests.map(x=>x.testResult).indexOf("PASSED") > -1)) {
+            if ((tests.map(x=>x.testResult).indexOf("FAILED") > -1)) {
+                buildResult = "PARTIAL-SUCCESS";
             } else {
-                isValid = false;
+                buildResult = "SUCCESS";
             }
+        } else {
+            buildResult = "FAILURE";
+        }
+        return buildResult;
+    }
 
-            // Parse benchmark variant
-            if ( ( curRegexResult = benchmarkVariantRegex.exec( curItr.value ) ) !== null ) {
-                curBenchmarkVariant = curRegexResult[1];
-            } else {
-                isValid = false;
+    static parsePerf(results) {
+        if(Array.isArray(results) && results.length > 0) {
+            for(let result of results) {
+                if (this.parse(result.testName, result.testOutput)) {
+                    const {benchmarkName, benchmarkVariant,testResult,testData} = this.parse(result.testName, result.testOutput);
```

Comment:
We don't need to call `parse` twice. We can should call `canParse` in the if statement, and then call `parse` inside that statement. `parse` is much heavier compared to `canParse`.

Reply:
Small spacing issue. Need to have a spaces among `benchmarkVariant,testResult,testData`.